samples/multi-agent/src/main/java/demo/multiagent/application/SessionMemoryConsumer.java

```diff
@@ -0,0 +1,80 @@
+package demo.multiagent.application;
+
+import akka.javasdk.agent.SessionMemoryEntity;
+import akka.javasdk.agent.SessionMemoryEntity.Event.AiMessageAdded;
+import akka.javasdk.agent.SessionMemoryEntity.Event.UserMessageAdded;
+import akka.javasdk.agent.SessionMessage;
+import akka.javasdk.annotations.ComponentId;
+import akka.javasdk.annotations.Consume;
+import akka.javasdk.client.ComponentClient;
+import akka.javasdk.consumer.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+@ComponentId("session-memory-eval-consumer")
+@Consume.FromEventSourcedEntity(value = SessionMemoryEntity.class, ignoreUnknown = true)
+public class SessionMemoryConsumer extends Consumer {
+
+  private static final Logger logger = LoggerFactory.getLogger(SessionMemoryConsumer.class);
+
+  private final ComponentClient componentClient;
+
+  public SessionMemoryConsumer(ComponentClient componentClient) {
+    this.componentClient = componentClient;
+  }
+
+  public Effect onUserMessage(UserMessageAdded event) {
+
+    return effects().done();
+  }
+
+  public Effect onAiMessage(AiMessageAdded event) {
+    if (event.componentId().equals("summarizer-agent")) {
+      evalToxicity(event);
+      evalSummarization(event);
+    }
+
+    return effects().done();
+  }
+
+  private void evalToxicity(AiMessageAdded event) {
+    componentClient
+        .forAgent()
+        .inSession(sessionId())
+        .method(ToxicityEvaluator::evaluate)
+        .invoke(event.message());
+  }
+
+  private void evalSummarization(AiMessageAdded event) {
+    var sessionHistory =
+        componentClient
+            .forEventSourcedEntity(sessionId())
+            .method(SessionMemoryEntity::getHistory)
+            .invoke(new SessionMemoryEntity.GetHistoryCmd(Optional.empty()));
+
+    Optional<SessionMessage> correspondingUserMessage = sessionHistory.messages().stream()
```

Comment:
perhaps we should have this as a direct method on the SessionMemoryEntity?

Reply:
Needing to do this in general is definitely more awkward, rather than evaluation frameworks where you'll get all the interactions in a session to evaluate.

Also more awkward with the evaluator needing to subscribe to all of the session memory events, and then filter down to which component ids it's actually interesting in.

Reply:
`corresponding that AiMessage sequence number.` ok so it's like: find me all messages between this AiMessage seq number and the first preceding UserMessage?

Reply:
yes